File: lc/py/051.py
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        def dfs(node):
            if node:
                dfs(node.left)
                dfs(node.right)
                node.left, node.right = node.right, node.left

        dfs(root)
        return root

File: lc/c/051.c
void dfs(struct TreeNode* node) {
    if (node) {
        dfs(node->left);
        dfs(node->right);
        struct TreeNode *tmp = node->left;
        node->left = node->right;
        node->right = tmp;
    }
}

struct TreeNode* invertTree(struct TreeNode* root){
    dfs(root);
    return root;
}
